java/src/jmri/util/FileUtilSupport.java

```diff
@@ -752,7 +794,7 @@ public String getScriptsPath() {
      *
      * @param path the scriptsPath to set
      */
-    public void setScriptsPath(String path) {
+    public void setScriptsPath(@Nonnull String path) {
```

Comment:
It needs to be possible to set this to null to reset it to the default calculated value. I think this possibility is leveraged when setting preferences.